File: integration-test/integration-test-java/src/main/java/com/lamblin/it/controller/DeleteController.java
/*
 * Lamblin
 * Copyright 2018 Borislav Shekerov
 * Licensed under Apache 2.0: https://github.com/BorislavShekerov/lamblin/blob/master/LICENSE
 */

package com.lamblin.it.controller;

import com.lamblin.core.model.HttpMethod;
import com.lamblin.core.model.HttpResponse;
import com.lamblin.core.model.annotation.Controller;
import com.lamblin.core.model.annotation.Endpoint;
import com.lamblin.core.model.annotation.PathParam;
import com.lamblin.core.model.annotation.QueryParam;
import com.lamblin.it.model.ResponseEntity;

import static com.lamblin.it.model.EndpointsKt.*;
import static com.lamblin.it.model.TestUtilsKt.*;
import static java.text.MessageFormat.format;

@Controller
public class DeleteController {

    @Endpoint(path = SIMPLE_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> simpleDeleteNoParams() {
        return HttpResponse.ok(new ResponseEntity(SIMPLE_DELETE_ENDPOINT));
    }

    @Endpoint(path = QUERY_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> singleQueryParamTest(@QueryParam(value = QUERY_PARAM_1) String queryParam) {
        return HttpResponse.ok(new ResponseEntity(format("{0}-{1}", QUERY_PARAM_DELETE_ENDPOINT, queryParam)));
    }

    @Endpoint(path = QUERY_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> multipleQueryParamTest(
            @QueryParam(QUERY_PARAM_1) String queryParam1,
            @QueryParam(QUERY_PARAM_2) String queryParam2) {

        return HttpResponse.ok(
                new ResponseEntity(
                        format(
                                "{0}-{1},{2}",
                                QUERY_PARAM_DELETE_ENDPOINT,
                                queryParam1,
                                queryParam2)));
    }

    @Endpoint(path = SINGLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> singlePathParamPath(
            @PathParam(PATH_PARAM_1) String pathParam) {

        return HttpResponse.ok(
                new ResponseEntity(
                        format(
                                "{0}-{1}",
                                SINGLE_PATH_PARAM_DELETE_ENDPOINT,
                                pathParam)));
    }

    @Endpoint(path = MULTIPLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> multiplePathParamPath(
            @PathParam(PATH_PARAM_1) String pathParamOne,
            @PathParam(PATH_PARAM_2) String pathParamTwo) {

        return HttpResponse.ok(
                new ResponseEntity(
                        format(
                                "{0}-{1},{2}",
                                MULTIPLE_PATH_PARAM_DELETE_ENDPOINT,
                                pathParamOne,
                                pathParamTwo)));
    }

    @Endpoint(path = MULTIPLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    public HttpResponse<ResponseEntity> multiplePathParamWithQueryParamsPath(
            @QueryParam(QUERY_PARAM_1) String queryParam,
            @PathParam(PATH_PARAM_1) String pathParamOne,
            @PathParam(PATH_PARAM_2) String pathParamTwo) {

        return HttpResponse.ok(
                new ResponseEntity(
                        format(
                                "{0}-{1},{2},{3}",
                                MULTIPLE_PATH_PARAM_DELETE_ENDPOINT,
                                queryParam,
                                pathParamOne,
                                pathParamTwo)));
    }

}

File: integration-test/integration-test-kotlin/src/main/kotlin/com/lamblin/it/controller/DeleteController.kt
package com.lamblin.it.controller

import com.lamblin.core.model.HttpMethod
import com.lamblin.core.model.HttpResponse
import com.lamblin.core.model.annotation.Controller
import com.lamblin.core.model.annotation.Endpoint
import com.lamblin.core.model.annotation.PathParam
import com.lamblin.core.model.annotation.QueryParam
import com.lamblin.it.model.*

@Controller
class DeleteController {

    @Endpoint(SIMPLE_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun simpleDeleteNoParams(): HttpResponse<ResponseEntity> {
        return HttpResponse.ok(ResponseEntity(SIMPLE_DELETE_ENDPOINT))
    }

    @Endpoint(QUERY_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun signleQueryParamTest(@QueryParam(QUERY_PARAM_1) queryParam: String): HttpResponse<ResponseEntity> {
        return HttpResponse.ok(ResponseEntity("$QUERY_PARAM_DELETE_ENDPOINT-$queryParam"))
    }

    @Endpoint(QUERY_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun multipleQueryParamTest(
        @QueryParam(QUERY_PARAM_1) queryParam1: String,
        @QueryParam(QUERY_PARAM_2) queryParam2: String
    ): HttpResponse<ResponseEntity> {

        return HttpResponse.ok(ResponseEntity("$QUERY_PARAM_DELETE_ENDPOINT-$queryParam1,$queryParam2"))
    }

    @Endpoint(SINGLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun singlePathParamPath(
        @PathParam(PATH_PARAM_1) pathParam: String
    ): HttpResponse<ResponseEntity> {

        return HttpResponse.ok(ResponseEntity("$SINGLE_PATH_PARAM_DELETE_ENDPOINT-$pathParam"))
    }

    @Endpoint(MULTIPLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun multiplePathParamPath(
        @PathParam(PATH_PARAM_1) pathParamOne: String,
        @PathParam(PATH_PARAM_2) pathParamTwo: String
    ): HttpResponse<ResponseEntity> {

        return HttpResponse.ok(
            ResponseEntity("$MULTIPLE_PATH_PARAM_DELETE_ENDPOINT-$pathParamOne,$pathParamTwo"))
    }

    @Endpoint(MULTIPLE_PATH_PARAM_DELETE_ENDPOINT, method = HttpMethod.DELETE)
    fun multiplePathParamWithQueryParamsPath(
        @QueryParam(QUERY_PARAM_1) queryParam: String,
        @PathParam(PATH_PARAM_1) pathParamOne: String,
        @PathParam(PATH_PARAM_2) pathParamTwo: String
    ): HttpResponse<ResponseEntity> {

        return HttpResponse.ok(
            ResponseEntity(
                "$MULTIPLE_PATH_PARAM_DELETE_ENDPOINT-$queryParam,$pathParamOne,$pathParamTwo"))
    }

}